WordPress/Classes/Services/MediaService.h

```diff
@@ -93,6 +93,30 @@
                     failure:(nullable void (^)(NSError * _Nonnull error))failure;
 
 /**
+ Deletes the Media object from the server. Note the Media is deleted, not trashed.
+
+ @param media object to delete.
+ @param success a block that will be invoked when the media deletion finished with success
+ @param failure a block that will be invoked when there is an error.
+ */
+- (void)deleteMedia:(nonnull Media *)media
+            success:(nullable void (^)())success
+            failure:(nullable void (^)(NSError * _Nonnull error))failure;
+
+/**
+ Deletes multiple Media object from the server. Note the Media objects are deleted, not trashed.
+
+ @param mediaObjects An array of media objects to delete.
+ @param progress a block that will be invoked after each media item is deleted successfully
+ @param success a block that will be invoked when the media deletion finished with success
+ @param failure a block that will be invoked when there is an error.
+ */
+- (void)deleteMultipleMedia:(nonnull NSArray<Media *> *)mediaObjects
```

Comment:
I went with `deleteMultiple...` to match the existing `updateMultiple...` method. Perhaps we could revisit this in the future (convert the whole lot to Swift? 😁)